case_studies/weak_lensing/lensing_metrics.py

```diff
@@ -6,35 +6,62 @@ class LensingMapMSE(Metric):
+        )
         self.add_state("convergence_sum_squared_err", default=torch.zeros(1), dist_reduce_fx="sum")
+        self.add_state(
+            "baseline_convergence_sum_squared_err", default=torch.zeros(1), dist_reduce_fx="sum"
+        )
         # potentially throws a division by zero error if true_idx is empty and uncaught
         self.add_state("total", default=torch.zeros(1), dist_reduce_fx="sum")
 
     def update(self, true_cat, est_cat, matching) -> None:
-        for i, match in enumerate(matching):
-            true_idx, est_idx = match
-            true_shear = true_cat["shear"][i, true_idx, :].flatten(end_dim=-2)
-            pred_shear = est_cat["shear"][i, est_idx, :].flatten(end_dim=-2)
-            true_convergence = true_cat["convergence"][i, true_idx, :].flatten()
-            pred_convergence = est_cat["convergence"][i, est_idx, :].flatten()
-
-            shear1_sq_err = ((true_shear[:, 0].flatten() - pred_shear[:, 0].flatten()) ** 2).sum()
-            shear2_sq_err = ((true_shear[:, 1].flatten() - pred_shear[:, 1].flatten()) ** 2).sum()
-            convergence_sq_err = ((true_convergence - pred_convergence) ** 2).sum()
-
-            self.total += true_idx.size(0)
-            self.shear1_sum_squared_err += shear1_sq_err
-            self.shear2_sum_squared_err += shear2_sq_err
-            self.convergence_sum_squared_err += convergence_sq_err
+        # along dim 2
+        true_shear = true_cat["shear"].flatten(1, 2)
+        pred_shear = est_cat["shear"].flatten(1, 2)
+        baseline_pred_shear = true_shear.mean(1).unsqueeze(1) * torch.ones_like(true_shear)
+        true_convergence = true_cat["convergence"].flatten(1, 2)
+        pred_convergence = est_cat["convergence"].flatten(1, 2)
+        baseline_pred_convergence = true_convergence.mean(1).unsqueeze(-1) * torch.ones_like(
+            true_convergence
+        )
+
+        shear1_sq_err = ((true_shear[:, :, 0] - pred_shear[:, :, 0]) ** 2).sum()
+        baseline_shear1_sq_err = ((true_shear[:, :, 0] - baseline_pred_shear[:, :, 0]) ** 2).sum()
+        shear2_sq_err = ((true_shear[:, :, 1] - pred_shear[:, :, 1]) ** 2).sum()
+        baseline_shear2_sq_err = ((true_shear[:, :, 0] - baseline_pred_shear[:, :, 1]) ** 2).sum()
+        convergence_sq_err = ((true_convergence - pred_convergence) ** 2).sum()
+        baseline_convergence_sq_err = ((true_convergence - baseline_pred_convergence) ** 2).sum()
+
+        self.shear1_sum_squared_err += shear1_sq_err
+        self.baseline_shear1_sum_squared_err += baseline_shear1_sq_err
+        self.shear2_sum_squared_err += shear2_sq_err
+        self.baseline_shear2_sum_squared_err += baseline_shear2_sq_err
+        self.convergence_sum_squared_err += convergence_sq_err
+        self.baseline_convergence_sum_squared_err += baseline_convergence_sq_err
+
+        self.total = torch.tensor(  # pylint: disable=attribute-defined-outside-init
```

Comment:
Would it work to set `self.total = None` in the constructor? (rather than disabling this check)